src/connector/src/sink/snowflake_redshift/redshift.rs

```diff
@@ -486,9 +568,71 @@ impl SinkCommitCoordinator for RedshiftSinkCommitter {
     async fn commit(
         &mut self,
         _epoch: u64,
-        _metadata: Vec<SinkMetadata>,
+        metadata: Vec<SinkMetadata>,
         add_columns: Option<Vec<Field>>,
     ) -> Result<()> {
+        let paths = metadata
+            .into_iter()
+            .filter(|m| {
+                if let Some(sink_metadata::Metadata::Serialized(SerializedMetadata { metadata })) =
+                    &m.metadata
+                {
+                    !metadata.is_empty()
+                } else {
+                    false
+                }
+            })
+            .map(|metadata| {
+                let path = if let Some(sink_metadata::Metadata::Serialized(SerializedMetadata {
+                    metadata,
+                })) = metadata.metadata
+                {
+                    String::from_utf8(metadata)
+                        .map_err(|e| SinkError::Config(anyhow!("Invalid UTF-8 in metadata: {}", e)))
+                } else {
+                    Err(SinkError::Config(anyhow!("Invalid metadata format")))
+                }?;
+                Ok(json!({
+                    "url": path,
+                    "mandatory": true
+                }))
+            })
+            .collect::<Result<Vec<_>>>()?;
+        if !paths.is_empty() {
+            let s3_operator = FileSink::<S3Sink>::new_s3_sink(&self.config.s3_inner)?;
+            let (mut writer, path) =
+                build_opendal_writer_path(&self.config.s3_inner, 0, &s3_operator, &None).await?;
+            let manifest_json = json!({
+                "entries": paths
+            });
+            let mut chunk_buf = BytesMut::new();
+            writeln!(chunk_buf, "{}", manifest_json).unwrap();
+            writer.write(chunk_buf.freeze()).await?;
+            writer
+                .close()
+                .await
+                .map_err(|e| SinkError::File(e.to_report_string()))?;
+            let table = if self.is_append_only {
+                &self.config.table
+            } else {
+                self.config.cdc_table.as_ref().ok_or_else(|| {
+                    SinkError::Config(anyhow!(
+                        "intermediate.table.name is required for non-append-only sink"
+                    ))
+                })?
+            };
+            let copy_into_sql = build_copy_into_sql(
+                self.config.schema.as_deref(),
+                table,
+                &path,
+                &self.config.s3_inner.access,
+                &self.config.s3_inner.secret,
+                &self.config.s3_inner.assume_role,
+            )?;
+            // run copy into
+            self.client.execute_sql_sync(&vec![copy_into_sql])?;
```

Comment:
How do we remove the files that are already merged into the redshift table? See my previous comment [here](https://github.com/risingwavelabs/risingwave/pull/22781#discussion_r2265574338).